src/main/java/com/nkh/ECommerceShop/service/ProductsService.java

```diff
@@ -40,7 +40,8 @@ public boolean checkProductStock(long productId, int requestedAmount){
     }
 
     @Transactional
-    public void deleteProduct(long id){
+    public boolean deleteProduct(long id){
```

Comment:
What is the reason behind using transactional here as you have only delete operation?

Reply:
2: get by id and delete